### PR TITLE
Fix external login texts

### PR DESCRIPTION
**Pull request by lariciamota**
## What's the purpose of this pull request?
This code was lost in the `next` branch

## How it works? 
Only show texts like "Entrar usando Google" or "Escolha uma forma de login" when they make sense/when store providers list exist

## How to test it?

See login page (in marin or boti switch to google/face to see the text)

### CODE REVIEW

File: packages/gatsby-theme-store/src/components/Auth/Providers/ExternalProvider/index.tsx [hunk 1]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable @typescript-eslint/no-extra-semi */
```
Comment (klzns) [inline packages/gatsby-theme-store/src/components/Auth/Providers/ExternalProvider/index.tsx:1]: Why is that needed?

**Review by klzns — APPROVED** [review]

File: packages/gatsby-theme-store/src/components/Auth/Providers/ExternalProvider/index.tsx (hunk 1 above)
Comment (lariciamota) [reply packages/gatsby-theme-store/src/components/Auth/Providers/ExternalProvider/index.tsx:1]: It was complaining about the `;(async` on line 36, but if I remove it complains about having to add a ; 🤔  And in both occasions it doesn't let me commit while I don't fix it

**Review by klzns — CHANGES_REQUESTED** [review]
The useEffect on line 36 should be changed. Currently it can break if the component is unmounted because of its async behaviour. Example: if the component unmounts when calling the setState('error') it will break.

Change to something like:

```tsx
useEffect(() => {
  let mounted = true
 
  async function login() {
    if (!window) return

    try {
      await startLogin({
        callbackUrl: oAuthCallbackUrl(),
      })

      window.location.href = oAuthRedirectUrl(providerName)
    } catch {
      if (!mounted) return

      setState('error')
    }
  }

  login()

  return () => {
    mounted = false
  }
}, [providerName, startLogin, state]
```

With that you can remove the eslint-ignore on the top of the file too

**Review by klzns — APPROVED** [review]
